### PR TITLE
Nodes which are a function of time

### PR DESCRIPTION
**Pull request by mundya**
As specified in #38. **Requires #53**

Nodes marked as a function of time are evaluated in advance and played back on the SpiNNaker board.

Greatest support is for periodic Nodes, though aperiodic Nodes are supported given a bounded simulation run time and sufficient memory.
- [ ] ~~Raise Exceptions if ineligible Nodes are marked~~ Some user intelligence will have to be displayed :unamused: 
- [ ] ~~Raise better Exceptions in the case of insufficient memory:~~ _ideally it would be nice to dynamically switch from on chip to on host in this case, but it's currently noticed too late in the build process.  Perhaps an intermediate representation between Nengo and PACMAN would be good after all..._
- [x] Test for Nodes with with multidimensional output
- [x] Rebase onto corrected Connection logic

### CODE REVIEW

**Comment by mundya** [comment]
Usage is as:

``` python
config = nengo_spinnaker.Config()

config[input].f_of_t = True
config[input].f_period = 2*np.pi  # None means aperiodic

sim = nengo_spinnaker.Simulator(model, config=config)
```

File: nengo_spinnaker/nodes/value_source_vertex.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        self.width = sum([t.size for t in self.transforms])
+        self.data_size = self._n_ticks * self.width
+        self.frames_per_block = 5*1024 / self.width  # 20KB / 4*t
+        self.full_blocks = self._n_ticks / self.frames_per_block
+        self.r_blocks = self._n_ticks % self.frames_per_block
+
+        transform = np.vstack(self.transforms)    # Combine the transforms
+        ts = np.arange(0, self._n_ticks * self.dt, self.dt)  # Eval points
+        vs = np.vectorize(self.node.output)(ts)   # Evaluate at each point
+        vs = vs.reshape((1, vs.size))
```
Comment (mundya) [inline nengo_spinnaker/nodes/value_source_vertex.py]: Test this for functions that return multiple dimensions.

**Comment by mundya** [comment]
I think we're good now,

``` python
import nengo
import nengo_spinnaker
import numpy as np

model = nengo.Network()

def feedin(t):
    return np.sin(t)

with model:
    a = nengo.Node(feedin, label='input')
    e = nengo.Ensemble(100, 2)
    f = nengo.Ensemble(100, 1)

    p = nengo.Probe(e, synapse=0.05)
    pf = nengo.Probe(f, synapse=0.05)

    nengo.Connection(a, e, transform=[[-.25], [.5]], function=lambda v: v if v > 0. else 0.)
    nengo.Connection(a, f)

# Configure `a` as being a function of time
config = nengo_spinnaker.Config()
config[a].f_of_t = True
config[a].f_period = 8*np.pi  # A lie!

# Create the simulation
sim = nengo_spinnaker.Simulator(model, config=config)
sim.run(20.)

from matplotlib import pyplot as plt
import mpltools.style
mpltools.style.use('ggplot')

plt.plot(sim.trange(), sim.data[p], label="E")
plt.plot(sim.trange(), sim.data[pf], label="F")
plt.xlabel('Time / s')
plt.legend()
plt.show()
```

results in:
fn_of_t

Matches Nengo (save for some noise).

**Comment by tcstewar** [comment]
Pretty!

**Comment by mundya** [comment]
(Yeah, I found `mpltools`...)

**Comment by tbekolay** [comment]
Also check out [seaborn](http://www.stanford.edu/~mwaskom/software/seaborn/) :smiley:

**Comment by mundya** [comment]
Also pretty!  I mostly used pgfplots (http://pgfplots.sourceforge.net/) beforehand so I'm pretty easy to please...

**Pull request merged**